src/main/java/com/powsybl/openloadflow/OpenSecurityAnalysis.java

```diff
@@ -171,36 +172,42 @@ SecurityAnalysisResult runSync(SecurityAnalysisParameters securityAnalysisParame
         return lfNetworks;
     }
 
-    private void detectViolations(LfNetwork network, LfContingency lfContingency, List<LimitViolation> violations) {
-        // detect violation limits on branches
-        List<LfBranch> branches = network.getBranches();
-        for (LfBranch branch : branches) {
-            if (lfContingency == null) {
-                detectBranchViolations(branch, violations);
-            } else if (!lfContingency.getBranches().contains(branch)) {
-                detectBranchViolations(branch, violations);
-            }
-        }
-        // detect violation limits on buses
+    /**
+     * Detect violations on branches and on buses
+     * @param branches branches on which the violation limits are checked
+     * @param buses buses on which the violation limits are checked
+     * @param violations list on which the violation limits encountered are added
+     */
+    private void detectViolations(Stream<LfBranch> branches, Stream<LfBus> buses, List<LimitViolation> violations) {
+        // Detect violation limits on branches
+        branches.forEach(b -> detectBranchViolations(b, violations));
+
+        // Detect violation limits on buses
         // TODO
-
     }
 
+    /**
+     * Detect violation limits on one branch and add them to the given list
+     * @param branch branch of interest
+     * @param violations list on which the violation limits encountered are added
+     */
     private void detectBranchViolations(LfBranch branch, List<LimitViolation> violations) {
-        // detect violation limits on a branch
-        double scale = 1;
-        if (branch.getBus1() != null && branch.getI1() > branch.getPermanentLimit1()) {
-            scale = PerUnit.SB / branch.getBus1().getNominalV();
-            LimitViolation limitViolation1 = new LimitViolation(branch.getId(), LimitViolationType.CURRENT, (String) null,
-                    2147483647, branch.getPermanentLimit1() * scale, (float) 0., branch.getI1() * scale, Branch.Side.ONE);
+        // Detect violation on the left
+        detectBranchViolationSide(branch.getId(), Branch.Side.ONE, branch.getBus1(), branch.getI1(), branch.getPermanentLimit1(), violations);
+
+        // Detect violation on the right
+        detectBranchViolationSide(branch.getId(), Branch.Side.TWO, branch.getBus2(), branch.getI2(), branch.getPermanentLimit2(), violations);
+    }
+
+    private void detectBranchViolationSide(String branchId, Branch.Side branchSide, LfBus busOfSide, double iOfSide,
+                                           double permanentLimitOfSide, List<LimitViolation> violations) {
+        if (busOfSide != null && iOfSide > permanentLimitOfSide) {
```

Comment:
Sorry, you made it here. So why do you need to increase the permanent limit getters ?

Reply:
To avoid NullPointerException before calling this method.